ui-performance/index.md

```diff
@@ -0,0 +1,484 @@
+
+1. Select **View > Tool Windows > Flutter Inspector**.
+1. In the toolbar, select the icon that looks like a bookshelf (<img src="images/performance-overlay-icon.png" alt="icon that resembles a bookshelf">).
+
+![IntelliJ Flutter Inspector Window](/images/intellij/visual-debugging.png)<br>
+
+The Flutter Inspector is available in the VS Code, Android Studio,
+and IntelliJ plugins, but its UI may change as it continues to evolve.
+The inspector is not currently available from the command line.
+Learn more about what the inspector can do in the
+[Flutter Widget Inspector](/inspector/) doc, as well as the
+[Flutter Inspector talk](https://www.youtube.com/watch?v=JIcmJNT9DNI)
+presented at DartConf 2018.
+
+In VS Code:
+
+1. Select **View > Command Palette…** to bring up the command palette.
+1. In the text field, enter "performance" and select
+   **Toggle Performance Overlay** from the list that pops up.
+   If this command isn’t available, make sure that that the app is running.
+
+#### Command line
+
+Toggle the performance overlay using the **P** key from the command line.
+
+#### Programmatically
+
+You can programmatically enable the PerformanceOverlay widget is by
+setting the `showPerformanceOverlay` property to `true` on the MaterialApp
+or WidgetsApp constructor:
+
+<!-- skip -->
+{% prettify dart %}
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      [[highlight]]showPerformanceOverlay: true,[[/highlight]]
+      title: 'My Awesome App',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyHomePage(title: 'My Awesome App'),
+    );
+  }
+}
+{% endprettify %}
+
+You are probably familiar with the Flutter Gallery example app.
+To use the performance overlay with Flutter Gallery,
+[download](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery),
+and run the app in profile mode. The program is written
+so that the app menu allows you to dynamically toggle the overlay,
+as well as enable checks for calls to `saveLayer` and the presence of
+cached images.
+
+<aside class="alert alert-info" markdown="1">
+**Note:** You can’t enable the performance overlay in the Flutter
+Gallery app downloaded from the App Store. That version of the app
+is compiled to release mode (not profile mode), and doesn’t provide
+a menu for enabling or disabling the overlay.
+</aside>
+
+### Identifying problems in the engine graph
+
+If the performance overlay shows red in the engine graph, start by profiling
+the engine (Dart VM), even if the rasterizer graph also shows red.
+To do this, use
+[Observatory](https://dart-lang.github.io/observatory/), Dart’s profiling tool.
+
+#### Displaying Observatory
+
+Observatory provides a statement-level single-stepping debugger for Dart and
+Flutter applications.  It also supports profiling, examining the heap,
+and reporting on code coverage. Observatory's _timeline_ view allows you
```

Comment:
I would combine these two sentences to just `The Observatory supports features like profiling, examining the heap, and displaying code coverage.`

Reply:
Done